core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -93,70 +121,130 @@ static class Scheduler {
     private static final long TIMEOUT = 100;
     private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
     private final BlockingQueue<Load> mLoadQueue = new LinkedBlockingQueue<>(CAPACITY);
-    private final AtomicInteger mCurrentSize = new AtomicInteger();
+    private final AtomicInteger mCurrentSize = new AtomicInteger(); // scheduler capacity indicator
     private final AtomicLong mIdGenerator = new AtomicLong();
+    private final FileSystemMaster mFileSystemMaster;
+    private final FileSystemContext mFileSystemContext;
+
+    public Scheduler(FileSystemMaster fileSystemMaster, FileSystemContext fileSystemContext) {
+      mFileSystemMaster = fileSystemMaster;
+      mFileSystemContext = fileSystemContext;
+    }
 
     void schedule(Load load)
-            throws ResourceExhaustedException, InterruptedException {
-      if (mCurrentSize.get() == CAPACITY) {
-        throw new ResourceExhaustedException(
-                "Insufficient capacity to enqueue load tasks!");
+            throws InterruptedException {
+      if (mCurrentSize.get() >= CAPACITY) {
+        waitForLoad();
```

Comment:
This will block the caller until a slot is freed up, so if load runs for hours, caller will not get a return for hours. It's probably not a good idea.

Reply:
changed to throw, thanks